src/capture.rs

```diff
@@ -813,6 +772,49 @@ pub mod sink {
 
 //     }
 
+// /// An iterator that yields with a `None` every so often.
+// pub struct YieldingIter<I> {
+//     /// When set, a time after which we should return `None`.
+//     start: Option<std::time::Instant>,
+//     after: std::time::Duration,
+//     iter: I,
+// }
+
+// impl<I> YieldingIter<I> {
+//     /// Construct a yielding iterator from an inter-yield duration.
+//     pub fn new_from(iter: I, yield_after: std::time::Duration) -> Self {
+//         Self {
+//             start: None,
+//             after: yield_after,
+//             iter,
+//         }
+//     }
+// }
+
+// impl<I: Iterator> Iterator for YieldingIter<I> {
+//     type Item = I::Item;
+//     fn next(&mut self) -> Option<Self::Item> {
+//         if self.start.is_none() {
+//             self.start = Some(std::time::Instant::now());
+//         }
+//         let start = self.start.as_ref().unwrap();
+//         if start.elapsed() > self.after {
+//             self.start = None;
+//             None
+//         } else {
+//             match self.iter.next() {
+//                 Some(x) => Some(x),
+//                 None => {
+//                     self.start = None;
+//                     None
+//                 }
+//             }
+//         }
+//     }
+// }
+
+
```

Comment:
I'd prefer removing this now commented implementation, or moving it to an example together with the kafka source.

Reply:
Me too! It's .. probably something that lives in `itertools` or the equivalent. Certainly not in any way specific to the code around here.